src/Stratis.Features.FederatedPeg/InputConsolidation/InputConsolidator.cs

```diff
@@ -0,0 +1,291 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Wallet.Interfaces;
+using Stratis.Bitcoin.Primitives;
+using Stratis.Bitcoin.Signals;
+using Stratis.Features.FederatedPeg.Events;
+using Stratis.Features.FederatedPeg.Interfaces;
+using Stratis.Features.FederatedPeg.Payloads;
+using Stratis.Features.FederatedPeg.TargetChain;
+using Stratis.Features.FederatedPeg.Wallet;
+
+namespace Stratis.Features.FederatedPeg.InputConsolidation
+{
+    public class InputConsolidator : IInputConsolidator
+    {
+        public static readonly Money ConsolidationFee = Money.Coins(0.005m); // 50 inputs. This is roughly half the same as fee on withdrawals
+
+        private readonly IFederatedPegBroadcaster federatedPegBroadcaster;
+        private readonly IFederationWalletTransactionHandler transactionHandler;
+        private readonly IFederationWalletManager walletManager;
+        private readonly IBroadcasterManager broadcasterManager;
+        private readonly IFederatedPegSettings settings;
+        private readonly ILogger logger;
+        private readonly Network network;
+
+        /// <summary>
+        /// Used to ensure only one operation is happening at a time.
```

Comment:
Please add remarks section and state which objects are protected by that lock. 
Objects that are protected should have a remarks section specifying which type of access should be protected by the lock